libcloudforensics/providers/gcp/internal/compute.py

```diff
@@ -417,6 +417,39 @@ def ListDiskByLabels(self,
     return self._ListByLabel(
         labels_filter, disk_service_object, filter_union)
 
+  def ListReservedExternalIps(self, zone: str) -> List[str]:
+    """Lists all static external IP addresses that are available to a zone.
+
+    The method first converts the zone to a region,
+    and then queries the GCE addresses resource.
+
+    Args:
+      zone (str): The zone in which the returned IPs would be available.
+
+    Returns:
+      List[str]: The list of available IPs in the specified zone.
+
+    Raises:
+      ValueError: If the zone is malformed.
+    """
+    # Convert zone to region
+    zone_parts = zone.split('-')
+    if len(zone_parts) != 3:
+      raise ValueError('Invalid zone: {0:s}.'.format(zone))
+    region = '-'.join(zone_parts[:-1])
+    # Request list of addresses
+    addresses_client = self.GceApi().addresses()
+    request = addresses_client.list(project=self.project_id, region=region)
+    response = request.execute()
+    ip_addresses = []
+    for address in response.get('items', []):
+      is_reserved = address['status'] == 'RESERVED'
+      is_external = address['addressType'] == 'EXTERNAL'
```

Comment:
you return an empty list [] if items not exist, then you try to access dict fields in address['status'] and address['addressType'], in case address is [] this will fail.
```suggestion
    for address in response.get('items', {}):
      is_reserved = address.get('status') == 'RESERVED'
      is_external = address.get('addressType') == 'EXTERNAL'
```

Reply:
I think here if `address` ends up being an empty list the `for` loop is skipped all-together

Reply:
Yes, the body of the loop will not be executed in the case that `'items'` is not present.